Rules/AvoidGlobalFunctions.cs

```diff
@@ -0,0 +1,205 @@
+
+            records = new List<DiagnosticRecord>();
+            this.fileName = fileName;
+
+            ast.Visit(this);
+
+            return records;
+        }
+
+        #region VisitComman functions
+        public override AstVisitAction VisitCommand(CommandAst commandAst)
+        {
+            if (!IsNewAliasCmdlet(commandAst))
+            {
+                return AstVisitAction.SkipChildren;
+            }
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitCommandParameter(CommandParameterAst commandParameterAst)
+        {
+            if(IsScopeParameterForNewAliasCmdlet(commandParameterAst))
+            {
+                if ((commandParameterAst.Argument != null)   // if the cmdlet looks like -Scope:Global check Parameter.Argument
+                    && (commandParameterAst.Argument.ToString().Equals("Global", StringComparison.OrdinalIgnoreCase)))
+                {
+                    records.Add(new DiagnosticRecord(
+                                    string.Format(CultureInfo.CurrentCulture, Strings.AvoidGlobalFunctionsAliasError),
+                                    commandParameterAst.Extent,
+                                    GetName(),
+                                    DiagnosticSeverity.Warning,
+                                    fileName,
+                                    commandParameterAst.ParameterName));
+                }
+                else
+                {
+                    var nextAst = FindNextAst(commandParameterAst);
+
+                    if((nextAst is StringConstantExpressionAst)   // if the cmdlet looks like -Scope Global
+                        && (((StringConstantExpressionAst)nextAst).Value.ToString().Equals("Global", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        records.Add(new DiagnosticRecord(
+                                string.Format(CultureInfo.CurrentCulture, Strings.AvoidGlobalFunctionsAliasError),
+                                ((StringConstantExpressionAst)nextAst).Extent,
+                                GetName(),
+                                DiagnosticSeverity.Warning,
+                                fileName,
+                                ((StringConstantExpressionAst)nextAst).Value));
+                    }
+                }
+            }
+
+            return AstVisitAction.SkipChildren;
+        }
+
+        public override AstVisitAction VisitFunctionDefinition(FunctionDefinitionAst functionDefinitionAst)
+        {
+            if (functionDefinitionAst.Name.StartsWith("Global:", StringComparison.OrdinalIgnoreCase) && IsModule())
+            {
+                records.Add(new DiagnosticRecord(
+                                string.Format(CultureInfo.CurrentCulture, Strings.AvoidGlobalFunctionsError),
+                                functionDefinitionAst.Extent,
```

Comment:
This is mark the entire function definition. Instead you can just mark the function name.  Have a look at [this](https://github.com/PowerShell/PSScriptAnalyzer/blob/bd7c1b1689febfbdefe2279b6b6ee365c521acac/Engine/Helper.cs#L789) method